src/bluesearch/server/search_server.py

```diff
@@ -110,32 +110,30 @@ def __init__(
 
         self.logger.info("Initialization done.")
 
-    def _get_model(self, model_name):
+    def _get_model(self, model_name: str) -> EmbeddingModel:
         """Construct an embedding model from its name.
 
         Parameters
         ----------
-        model_name : str
+        model_name :
             The name of the model.
 
         Returns
         -------
-        bluesearch.embedding_models.EmbeddingModel
-            The embedding model class.
+        sentence_embedding_model :
+            The sentence embedding model instance.
         """
         biobert_nli_sts_cord19_v1_model_name = "biobert_nli_sts_cord19_v1"
         biobert_nli_sts_cord19_v1_model_path = (
             self.trained_models_path / biobert_nli_sts_cord19_v1_model_name
         )
 
         model_factories = {
-            "SBioBERT": lambda: SBioBERT(),
-            "SBERT": lambda: SentTransformer("bert-base-nli-mean-tokens"),
-            "BIOBERT NLI+STS": lambda: SentTransformer(
-                "clagator/biobert_v1.1_pubmed_nli_sts"
-            ),
-            "BioBERT NLI+STS CORD-19 v1": lambda: SentTransformer(
-                biobert_nli_sts_cord19_v1_model_path
+            "SBioBERT": lambda: get_embedding_model("SBioBERT"),
+            "SBERT": lambda: get_embedding_model("SBERT"),
+            "BioBERT NLI+STS": lambda: get_embedding_model("BioBERT NLI+STS"),
+            "BioBERT NLI+STS CORD-19 v1": lambda: get_embedding_model(
+                "SentTransformer", checkpoint_path=biobert_nli_sts_cord19_v1_model_path
             ),
         }
```

Comment:
Note as a reminder:

I feel like we are almost to https://github.com/BlueBrain/Search/issues/286 but there is still an extra step to do.

So, https://github.com/BlueBrain/Search/issues/286 is still to do (not the goal of the current PR).

Reply:
Yes I was thinking about the same.